src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1640,66 +1494,150 @@ ExtractInsertPartitionValue(Query *query, Var *partitionColumn)
 /* RouterSelectJob builds a Job to represent a single shard select query */
 static Job *
 RouterSelectJob(Query *originalQuery, RelationRestrictionContext *restrictionContext,
```

Comment:
Maybe `PlanRouterJob`, also update the comment to say that it handles UPDATE and DELETE as well.